### PR TITLE
Removed insecure use of `open` in sign.rb

### PR DESCRIPTION
**Pull request by Br3nda**
* Removed the brakeman ignore file
* added faraday
* fixes our security mess with the use of `open` to do http requests.
* moves Image Processor from models to services, because it is a service. Then brakeman stops thinking it's a model
* Fix code coverage reporting

### CODE REVIEW

**Comment by Br3nda** [comment]
i suspect the `k` variable needs escaping in here too, but would love a second opinion

```
      if v.is_a?(Array)
        v.each { |ea| query_string << "#{k}=#{CGI.escape(ea.to_s)}" if ea.present? }
      elsif v.present?
        query_string << "#{k}=#{CGI.escape(v.to_s)}"
      end
```

**Comment by Br3nda** [comment]
it's not following the http 302 on request for a random one. I'll fix

File: app/models/sign.rb [hunk 1]
```diff
@@ -76,26 +75,20 @@ def self.search(params)
...
-
-  def self.url_for_search(query)
+  def self.uri_for_search(query)
     # The handling of arrays in query strings is different
     # in the API than in rails
     return SIGN_URL unless query.is_a?(Hash)
+    '?' + _query_string_for_search(query).join('&')
+  end
+
+  def self._query_string_for_search(query)
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/models/sign.rb]: I've reduced this from 11.

**Comment by Br3nda** [comment]
I've moved the ImageProcessor to services folder, so brakeman doesn't think it's a model.

File: app/models/sign.rb [hunk 2]
```diff
@@ -76,26 +75,20 @@ def self.search(params)
   end
 
   def self.xml_request(params)
-    xml_document = nil
-    url = url_for_search(params)
-    time = Benchmark.measure do
-      xml_document = Nokogiri::XML(open(url))
-    end
+    xml_document = Nokogiri::XML(http_conn.get(uri_for_search(params)).body)
```
Comment (elspeth-rabid) [inline app/models/sign.rb:78]: Yay! No more slow queries :-)

File: app/models/sign.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (elspeth-rabid) [reply app/models/sign.rb]: What is the leading underscore convention?

File: app/models/sign.rb [hunk 3]
```diff
@@ -104,6 +97,13 @@ def self.url_for_search(query)
         query_string << "#{k}=#{CGI.escape(v.to_s)}"
```
Comment (elspeth-rabid) [inline app/models/sign.rb:97]: Well... the keys are not user provided but easily tampered with... I'd say escape em!

File: spec/models/sign_spec.rb [hunk 4]
```diff
@@ -52,12 +52,12 @@
   end
 
   describe '.random' do
-    it 'retrieves two random signs' do
-      sign_one = Sign.random
-      sign_two = Sign.random
-      expect(sign_one.is_a?(Sign)).to eq(true)
-      expect(sign_two.is_a?(Sign)).to eq(true)
-      expect(sign_one.id).not_to eq(sign_two.id)
+    describe 'retrieves two random signs' do
```
Comment (elspeth-rabid) [inline spec/models/sign_spec.rb:55]: Nicer!

File: app/models/sign.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/models/sign.rb]: oops. I've been coding in python more than ruby. It's used to mark a method as private.

Ruby has it's own way.

I think some of these class methods could go into a service instead. It's pretending to be ActiveRecord, but it's not.

File: app/models/sign.rb (hunk 3 above)
Comment (Br3nda) [reply app/models/sign.rb:97]: if you follow it back up through the stack, the keys do come from `params`

File: app/models/sign.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (elspeth-rabid) [reply app/models/sign.rb]: I do agree, we will certainly be refactoring model and controller stuff into services when we do touch files.
I think the underscore should possibly go.

File: app/models/sign.rb (hunk 3 above)
Comment (elspeth-rabid) [reply app/models/sign.rb:97]: Yeah, but with GET right? So the params are sent in the URL and can be tampered. Or am I missing the point? Ha ha.

**Comment by Br3nda** [comment]
They're sent in a http GET to FreeLex - so yes. They could be something
malicious.

File: app/models/sign.rb (hunk 3 above)
Comment (Br3nda) [reply app/models/sign.rb:97]: They're sent in a http GET to FreeLex - so yes. They could be something malicious.

**Review by ghostPepperWonton — APPROVED** [review]

**Pull request merged**